Expose WordPress helper manifest env to Lab Node rig workloads

## Problem
After Homeboy gained declared rig package dependencies, the SSI fixture matrix now installs and materializes its shared helper on Lab, but the Node workload cannot resolve WordPress extension helpers.

Run: `ssi-matrix-latest-20260626-r5`

Persisted bench run:

`homeboy runs show 8967a22f-ad5b-4197-9256-9a3df4c4b987`

Runner job:

`homeboy runner job logs homeboy-lab 1b72a4f2-87bd-4c41-b5fc-46b2b5eeaa74`

Failure:

```text
WORKLOAD_ERROR: static-site-fixture-matrix — iteration 1/1 threw: Cannot find module 'homeboy-extension-wordpress/wp-codebox-recipe-helper'
Require stack:
- /home/chubes/Developer/_lab_workspaces/homeboy-rigs-ssi-matrix-run-latest-.../shared/wp-codebox/recipe.mjs
```

The shared helper first checks:

- `HOMEBOY_WP_CODEBOX_RECIPE_HELPER`
- `HOMEBOY_WORDPRESS_HELPER_MANIFEST`
- fallback package import: `homeboy-extension-wordpress/wp-codebox-recipe-helper`

The runner has the WordPress extension installed and ready:

`/home/chubes/.config/homeboy/extensions/wordpress`

But the Node rig workload environment does not expose `HOMEBOY_WORDPRESS_HELPER_MANIFEST`, so the helper falls back to a package import that is not resolvable from the materialized rig workspace.

## Expected
Homeboy/Lab Node rig workloads should receive stable helper discovery env for installed Homeboy extensions, at minimum:

- `HOMEBOY_WORDPRESS_HELPER_MANIFEST=/home/chubes/.config/homeboy/extensions/wordpress/lib/helper-manifest.js`

or an equivalent generic extension-helper env mechanism.

The fix should avoid per-rig shims. Rigs should be able to compose reusable WordPress extension helpers from Node workloads on Lab the same way they do locally.

## Acceptance criteria
- Lab Node rig workloads can load WordPress extension helpers via `HOMEBOY_WORDPRESS_HELPER_MANIFEST`.
- SSI fixture matrix reaches the WP Codebox recipe execution phase instead of failing module resolution.
- The env is assembled through Homeboy/Homeboy Extensions runtime plumbing, not hardcoded into the SSI rig.
- Missing extension/helper state should produce a clear diagnostic.

## Related
- #6574: declared rig package dependencies for Lab materialization.
- #6578 / #6579: package dependency support and corrected boundary validation.

## AI assistance
- **AI assistance:** Yes
- **Tool(s):** gpt-5.5 via OpenCode
- **Used for:** Capturing the Lab failure and identifying missing extension helper env wiring.

--- a/wordpress/package.json
+++ b/wordpress/package.json
@@ -100,6 +100,7 @@
     "test:full-surface-coverage": "node tests/full-surface-coverage-smoke.js",
     "test:wordpress-fuzz-coverage-aggregate": "node tests/wordpress-fuzz-coverage-aggregate-smoke.js",
     "test:external-http-guardrail": "node tests/external-http-guardrail-smoke.js",
+    "test:env-provider": "node tests/env-provider-smoke.js",
     "test:wordpress-helper-consumer": "node tests/wordpress-helper-consumer-smoke.mjs",
     "test:materialized-site-quality": "node tests/materialized-site-quality-smoke.js",
     "test:timing-correlator": "node tests/timing-correlator-smoke.js",

--- a/wordpress/scripts/env-provider.js
+++ b/wordpress/scripts/env-provider.js
@@ -1,0 +1,16 @@
+#!/usr/bin/env node
+'use strict';
+
+const fs = require('node:fs');
+const path = require('node:path');
+
+const extensionRoot = path.resolve(__dirname, '..');
+const helperManifest = path.join(extensionRoot, 'lib', 'helper-manifest.js');
+
+const env = {};
+
+if (fs.existsSync(helperManifest)) {
+	env.HOMEBOY_WORDPRESS_HELPER_MANIFEST = helperManifest;
+}
+
+process.stdout.write(`${JSON.stringify(env)}\n`);

--- a/wordpress/tests/env-provider-smoke.js
+++ b/wordpress/tests/env-provider-smoke.js
@@ -1,0 +1,21 @@
+'use strict';
+
+const assert = require('node:assert');
+const childProcess = require('node:child_process');
+const path = require('node:path');
+
+const extensionRoot = path.resolve(__dirname, '..');
+const provider = path.join(extensionRoot, 'scripts', 'env-provider.js');
+const output = childProcess.execFileSync(process.execPath, [provider], {
+	cwd: extensionRoot,
+	encoding: 'utf8',
+});
+const env = JSON.parse(output);
+
+assert.deepEqual(Object.keys(env), ['HOMEBOY_WORDPRESS_HELPER_MANIFEST']);
+assert.equal(
+	env.HOMEBOY_WORDPRESS_HELPER_MANIFEST,
+	path.join(extensionRoot, 'lib', 'helper-manifest.js')
+);
+
+console.log('env provider smoke passed');

--- a/wordpress/wordpress.json
+++ b/wordpress/wordpress.json
@@ -4,6 +4,9 @@
   "icon": "w.circle.fill",
   "description": "WordPress project type with WP-CLI integration",
   "author": "Extra Chill",
+  "env_provider": {
+    "script": "scripts/env-provider.js"
+  },
   "provides": {
     "file_extensions": ["php", "css", "ts", "tsx", "js", "jsx", "json"],
     "capabilities": ["fingerprint", "refactor", "crossref", "validate", "fuzz"]

